src/i3_json_parser/i3_json_parser-utility.cpp

```diff
@@ -270,6 +298,7 @@ i3_containers::node i3_json_parser::extract_tree(const rapidjson::Value& a_json_
     {
         node.floating_nodes.push_back(extract_tree(node_object));
     }
+    node.marks = extract_marks(a_json_object);
```

Comment:
I still feel like this one should be called with the similar functions above (on line 276).
Although it doesn't change the overall behavior, it feels strange to compute everything for current node, then go into recursion for tilling/floating and only afterwards compute marks for the current node.